tidy3d/components/source.py

```diff
@@ -291,7 +301,72 @@ def amp_time(self, time: float) -> complex:
         return const * offset * oscillation * amp
 
 
-SourceTimeType = Union[GaussianPulse, ContinuousWave]
+class CustomSourceTime(Pulse):
+    """Custom source time dependence.
```

Comment:
I think this and the note require a bit more discussion. So if a user makes a CustomTimeSource with some arbitrary values `v_i` placed at times `t_i`, the source that the simulation sees at time `t` will just be the `v_i` corresponding to the nearest time coordinate? Wouldn't this have the potential to cause a source time dependence that isn't smooth (and therefore contains very high frequency components?). If so, is there a way to mitigate this? (eg by providing a `dt` that is much smaller than the simulation `dt`?  I just think we need to make it explicit what this object is doing on the solver side, eg how the source value is chosen at time `t` given the data points, and also perhaps discuss these edge cases (like where the data points are not the same as the simulation time) and potential issues.

Reply:
yeah, using a small `dt` migitages this. Agreed, need more discussion here. I'll also split off the PR for CustomSourceTime